### PR TITLE
ABE Automation MVP

### PR DESCRIPTION
**Pull request by hshaosf**
# Automation Logic
* When submission is in `Submitted` status
    * If record is found in `IDADATA`
        * If there is existing classification in `IDADATA`
            * Update submission to `Existing classification` status
        * Else
            * Update `SPOT_CHECK_PERCENT` of submission to `Spot check` status
            * Otherwise if submission has `IDA_CATEGORY_ID_LOOKUP_id`
                * Update submission to `Recorded`
* When submission is in `Recorded` status
    * If record is found in `IDADATA`
        * If record has have `IDA_CATEGORY_ID`
            * If record `IDA_CATEGORY_ID` does not match submission `IDA_CATEGORY_ID_LOOKUP_id`
                * Update submission to `Existing classification` status
        * Else
            * Update `IDA_CATEGORY_ID`, `LAST_MODIFIED_BY`, `LAST_MODIFIED_DATE` in `IDADATA` for record
            * Generate a new `IDA_APP_NUM` if it does not have one.

### CODE REVIEW

File: webhook/__init__.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    # pylint: disable=too-many-nested-blocks
+    try:
+        validate_access(req)
+
+        response = Response()
+        if req.get_body() and len(req.get_body()):
+            response.status_code = 202
+            # pylint: disable=protected-access
+            response._content = b'"202 Accepted"'
```
Comment (josh-chou) [inline webhook/__init__.py]: Is there some lookup for this?

https://stackoverflow.com/questions/54999485/converty-http-response-code-to-text-message-with-python-standard-package

File: shared_code/noco.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+def get_by_id(table, idn):
+    """ get record by id number """
+    noco_api = os.getenv('NOCO_API_URL')
+    noco_request_url = f"{noco_api}{table}/{idn}"
+    print(noco_request_url)
+    response = requests.get(
+        noco_request_url,
+        headers={'xc-auth': os.getenv('NOCO_API_KEY')}
+    )
```
Comment (josh-chou) [inline shared_code/noco.py:13]: maybe call response.raise_for_status()

File: shared_code/noco.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+def update_by_id(table, idn, fields):
+    """ Update record by id """
+    noco_api = os.getenv('NOCO_API_URL')
+    noco_request_url = f"{noco_api}{table}/{idn}"
+    print(fields)
+    response = requests.put(
+        noco_request_url,
+        headers={'xc-auth': os.getenv('NOCO_API_KEY')},
+        json=fields
+    )
```
Comment (josh-chou) [inline shared_code/noco.py:26]: maybe call response.raise_for_status()

File: webhook/__init__.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+                                            str(datetime.now(pytz.timezone('US/Pacific')))
+                                    }
+                                    print(record_json["IDA_CATEGORY_ID"])
+                                    print(updates)
+                                    record = noco.update_by_id(
+                                        "IDADATA", record_json["ID"], updates)
+
+                                    print(record.content)
+                    response.status_code = 200
+                    response._content = b'"200 TEST"'
```
Comment (josh-chou) [inline webhook/__init__.py]: 200 OK

**Review by josh-chou — APPROVED** [review]
Just some minor suggestions.  Looking good!

File: shared_code/noco.py (hunk 3 above)
Comment (hshaosf) [reply shared_code/noco.py:26]: https://github.com/SFDigitalServices/abe-automation-fn-py/pull/1/commits/1989fedcaeb8266ad1aa296feb4ee81e0205f4af

File: shared_code/noco.py (hunk 2 above)
Comment (hshaosf) [reply shared_code/noco.py:13]: https://github.com/SFDigitalServices/abe-automation-fn-py/pull/1/commits/1989fedcaeb8266ad1aa296feb4ee81e0205f4af

File: webhook/__init__.py (hunk 1 above)
Comment (hshaosf) [reply webhook/__init__.py]: https://github.com/SFDigitalServices/abe-automation-fn-py/pull/1/commits/32b413e0edaaf2dc628f2a6f3b58888a464c0660

**Review by josh-chou — COMMENTED** [review]
@hshaosf can you review these updates?  can't add you as a reviewer - probably because you create the pr?

**Comment by hshaosf** [comment]
> @hshaosf can you review these updates? can't add you as a reviewer - probably because you create the pr?

@josh-chou  LGTM! 👍

**Comment by hshaosf** [comment]
> @hshaosf can you review these updates? can't add you as a reviewer - probably because you create the pr?

@josh-chou  LGTM! 👍

**Pull request merged**